### ISSUE TITLE
Ui - Date component - Mass edit - Sticky listing issue

### ISSUE THREAD

**Issue by Nuranto** [magento/magento2#37447]
### Preconditions and environment

- Magento version 2.4.6

### Steps to reproduce

1. Create a listing with inline edit, mass edit, and an editable date column
2. Set the listing sticky
3. Try to mass edit date via date picker

### Expected result

Date input is filled with the correct selected date.

### Actual result

Date input stays empty.

### Additional information

If you remove the sticky option, it works.

Here is a listing you can use to reproduce quickly : 

```
<?xml version="1.0"?>
<listing xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:module:Magento_Ui:etc/ui_configuration.xsd">
    <argument name="data" xsi:type="array">
        <item name="js_config" xsi:type="array">
            <item name="provider" xsi:type="string">test_listing.product_listing_data_source</item>
        </item>
    </argument>
    <settings>
        <spinner>test_columns</spinner>
        <deps>
            <dep>test_listing.product_listing_data_source</dep>
        </deps>
    </settings>
    <dataSource name="product_listing_data_source" component="Magento_Ui/js/grid/provider">
        <settings>
            <storageConfig>
                <param name="indexField" xsi:type="string">entity_id</param>
            </storageConfig>
            <updateUrl path="mui/index/render"/>
        </settings>
        <aclResource>Magento_Catalog::products</aclResource>
        <dataProvider class="Magento\Catalog\Ui\DataProvider\Product\ProductDataProvider" name="product_listing_data_source">
            <settings>
                <requestFieldName>id</requestFieldName>
                <primaryFieldName>entity_id</primaryFieldName>
            </settings>
        </dataProvider>
    </dataSource>
    <listingToolbar name="listing_top">
        <settings>
            <sticky>true</sticky>
        </settings>
        <bookmark name="bookmarks"/>
        <columnsControls name="columns_controls"/>
        <filterSearch name="fulltext"/>
        <exportButton name="export_button" />
        <filters name="listing_filters">
            <settings>
                <storageConfig>
                    <param name="provider" xsi:type="string">test_listing.test_listing.listing_top.bookmarks</param>
                    <param name="namespace" xsi:type="string">current.filters</param>
                </storageConfig>
                <childDefaults>
                    <param name="provider" xsi:type="string">test_listing.test_listing.listing_top.listing_filters</param>
                    <param name="imports" xsi:type="array">
                        <item name="visible" xsi:type="string">test_listing.test_listing.listing_top.bookmarks:current.columns.${ $.index }.visible</item>
                    </param>
                </childDefaults>
            </settings>
        </filters>
        <massaction name="listing_massaction">
            <action name="edit">
                <settings>
                    <callback>
                        <target>editSelected</target>
                        <provider>test_listing.test_listing.test_columns_editor</provider>
                    </callback>
                    <type>edit</type>
                    <label translate="true">Edit</label>
                </settings>
            </action>
        </massaction>
        <paging name="listing_paging"/>
    </listingToolbar>
    <columns name="test_columns">
        
        <settings>
            <childDefaults>
                <param name="fieldAction" xsi:type="array">
                    <item name="provider" xsi:type="string">test_listing.test_listing.test_columns_editor</item>
                    <item name="target" xsi:type="string">startEdit</item>
                    <item name="params" xsi:type="array">
                        <item name="0" xsi:type="string">${ $.$data.rowIndex }</item>
                        <item name="1" xsi:type="boolean">true</item>
                    </item>
                </param>
            </childDefaults>
            <editorConfig>
                <param name="clientConfig" xsi:type="array">
                    <item name="saveUrl" xsi:type="url" path="test/test/test"/>
                    <item name="validateBeforeSave" xsi:type="boolean">false</item>
                </param>
                <param name="indexField" xsi:type="string">entity_id</param>
                <param name="enabled" xsi:type="boolean">true</param>
                <param name="selectProvider" xsi:type="string">test_listing.test_listing.test_columns.ids</param>
            </editorConfig>
        </settings>

        <selectionsColumn name="ids">
            <settings>
                <indexField>entity_id</indexField>
                <visible>true</visible>
            </settings>
        </selectionsColumn>
        
        <column name="sku">
            <settings>
                <filter>text</filter>
                <dataType>text</dataType>
                <label translate="true">SKU</label>
                <editor>
                    <editorType>text</editorType>
                </editor>
            </settings>
        </column>
        <column name="news_from_date">
            <settings>
                <timezone>false</timezone>
                <dateFormat>d MMM y</dateFormat>
                <filter>dateRange</filter>
                <dataType>date</dataType>
                <label translate="true">News From date</label>
                <editor>
                    <editorType>date</editorType>
                </editor>
            </settings>
        </column>

    </columns>
</listing>
```

The same listing works if you remove : 
```
<settings>
            <sticky>true</sticky>
        </settings>
```

### Triage and priority

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [X] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by engcom-Hotel** [comment]
Hello @Nuranto,

Thanks for the report and collaboration!

We have tried to reproduce the issue in the vanilla Magento 2.4-develop branch but it seems the issue is not reproducible for us.

We have made the below custom module to reproduce the issue:
[Dev.zip](https://github.com/magento/magento2/files/13545395/Dev.zip)

We have referred the below devdocs to reproduce the issue:

https://developer.adobe.com/commerce/php/development/components/add-admin-grid/

Let us know if we have missed anything.

Thanks
